web_monitoring/db.py

```diff
@@ -760,3 +781,25 @@ def get_version_by_versionista_id(self, versionista_id):
         # Make result look like the result of `get_version` rather than the
         # result of `list_versions`.
         return {'data': result['data'][0]}
+
+    def validate_db_credentials(self):
+        """
+        Validate that the DB Client is authorized for the provided host
+
+        Parameters
+        ----------
+        none
+
+        Returns
+        -------
+        none
+        """
+        try:
+            response = self.get_user_session()
+        except requests.exceptions.HTTPError as exc:
+            if exc.response.status_code == 401:
+                raise MissingCredentials(f'\n\n\n\nCheck your DB credentials - 401 Unauthorized for url: {exc.request.url}\n\n\n\n')
```

Comment:
This logic is probably better handled in `_process_errors`, since it really applies to all requests. It also might be better if this error were a little less “presentational” — without all the line breaks (and if we really want them, use a multi-line string in stead of `\n`). `cli.py` is a better place to get fancy with formatting.

This should probably also be `raise InvalidCredentials(...)`, right?

Reply:
Makes sense. What's your general thought on getting fancy with the formatting in `cli.py`? 👍 or 👎?

Reply:
👍 Nice, explanatory, user-centric error messages are always good when surfaced from the script/CLI level! (Also good lower down; they should just be more focused more on developer rather than user ergonomics.)

Reply:
Makes sense - moved the 'presentational' error handling to the cli and used a docstring instead of line breaks.